org-formation/050-costs/program-code-rules.json

```diff
@@ -0,0 +1,398 @@
+[
+  {
+    "Type": "REGULAR",
+    "Value": "990300 Platform Infrastructure",
+    "Rule": {
+      "And": [
+        {
+          "Dimensions": {
+            "Key": "LINKED_ACCOUNT",
+            "Values": [
+              "531805629419",
+              "140124849929",
+              "254940560790",
+              "231505186444",
+              "153370007719",
+              "087547153226",
+              "797640923903",
+              "867686887310",
+              "420786776710",
+              "745159704268",
+              "325565585839",
+              "383874245509",
+              "449435941126",
+              "649232250620",
+              "804034162148",
+              "743644221192"
+            ],
+            "MatchOptions": [
+              "EQUALS"
+            ]
+          }
+        },
+        {
+          "Tags": {
+            "Key": "CostCenter",
+            "MatchOptions": [
+              "ABSENT"
+            ]
+          }
+        },
+        {
+          "Tags": {
+            "Key": "CostCenterOther",
+            "MatchOptions": [
+              "ABSENT"
+            ]
+          }
+        }
+      ]
+    }
+  },
+  {
+    "Type": "REGULAR",
```

Comment:
What is the purpose of `CostCenter` tag matching rules for specific tag values?  Isn't this handled by the generic 'inheritance' rule at the bottom of the file?
```
  {
    "Type": "INHERITED_VALUE",
    "InheritedValue": {
      "DimensionName": "TAG",
      "DimensionKey": "CostCenter"
    }
  }
```

Reply:
The purpose is to group multiple tag values into a single cost category. In some cases we have trivial differences to group, like ignoring the difference between a `/` and a `_`; and in some other cases we want to combine subprojects into a single category, like grouping 301100, 301101, and 301102 into a single cost category. And in any case where we are creating a category with a regular rule rather than inheriting a value, we have to remove punctuation from the name.

Changing our tagging format to put the code at the beginning of the string and enforcing that format would make the rules quite a bit simpler because we could just use a STARTS_WITH match and match against the significant digits of the code.